test/powershell/engine/DefaultCommands.Tests.ps1

```diff
@@ -528,4 +528,225 @@ Describe "Verify approved aliases list" -Tags "CI" {
+                if ($UsePowerShell)
+                {
+                    # avoid expression evaluation
+                    $null = $sb.AppendLine('@"')
+                }
+                $null = $sb.Append($Expression)
+                if ($UsePowerShell)
+                {
+                    $null = $sb.AppendLine()
+                    $null = $sb.Append('"@')
+                }
+            }
+            $scriptBlock = [ScriptBlock]::Create($sb.ToString())
+
+            # build the command line arguments that the function should pass to sudo
+            $null = $sb.Clear()
+            if (-not [string]::IsNullOrEmpty($sudoArgs))
+            {
+                $null = $sb.Append($sudoArgs)
+            }
+
+            if (-not [string]::IsNullOrEmpty($Expression))
+            {
+                if ($sb.Length -gt 0)
+                {
+                    $null = $sb.Append(' ')
+                }
+                if ($UsePowerShell)
+                {
+                    $bytes = [System.Text.Encoding]::Unicode.GetBytes($Expression)
+                    $arguments = [Convert]::ToBase64String($bytes)
+                    $null = $sb.AppendFormat('powershell -encodedCommand {0}', $arguments)
+                }
+                else
+                {
+                    $null = $sb.Append($Expression)
+                }
+            }
+            # invoke the sudo function
+            $output = & $scriptBlock
+            # return the expected and actual arguments passed to sudo
+            return @{ExpectedArguments = $sb.ToString(); ActualArguments = $output}
+        }
+
+        It "Validates sudo parameters formatting using the mock" -Skip:$skipTest {
+            $result = InvokeSudo -SudoArgs '--verify'
+            $result.ActualArguments | Should Be $result.ExpectedArguments
+        }
+
+        It "Validates sudo parameters and expression formatting using the mock" -Skip:$skipTest {
+            $result = InvokeSudo -SudoArgs '--verify' -Expression 'ls /etc'
+            $result.ActualArguments | Should Be $result.ExpectedArguments
+        }
+
+        It "Validates native command formatting using the mock" -Skip:$skipTest {
+            $result = InvokeSudo -Expression 'ls /etc'
+            $result.ActualArguments | Should Be $result.ExpectedArguments
+        }
+
+        It "Validates powershell command formatting using the mock" -Skip:$skipTest {
+            $expression = 'get-item /etc | select-object -Property FullName'
+            $result = InvokeSudo -Expression $expression -UsePowerShell
+            $result.ActualArguments | Should Be $result.ExpectedArguments
+        }
+    }
+
+    It "Validates sudo -V is not interpreted as -Verbose" -Skip:$skipTest {
+        $expectedValues = @(
+            'Sudo version',
+            'Sudoers policy plugin version',
+            'Sudoers file grammar version',
+            'Sudoers I/O plugin version'
+            )
+        $result = (sudo -V)

```

Comment:
if you want to be absolutely certain that you're getting your function, you can use the construct
`& ${function:sudo} --V`